app/src/js/controllers/ImageController.js

```diff
@@ -47,6 +48,44 @@ angular.module('media_manager')
       CourseCache.current_image = CourseCache.images[ic.index];
       resetBreadcrumb();
     }
-  }
+  };
+
+  ic.newLabel = '';
+  ic.newValue = '';
+  ic.saveMetadata = function(label, value){
+    if(label !== undefined && value !== undefined){
+      ic.image.metadata.forEach(function(item, index, arr){
+        if(item.label == label){
+          arr[index].label = label;
+          arr[index].value = value;
+          Image.update({}, ic.image, function success(data){
```

Comment:
I guess I worry about using indexes -- I imagine some scenario where the list is sorted and the indexes from the `ng-repeat` don't match the indexes of the actual array